tracer/src/Datadog.Trace.ClrProfiler.Managed.Loader/StartupLogger.cs

```diff
@@ -84,7 +84,22 @@ private static string GetLogDirectory()
                 {
                     if (Environment.OSVersion.Platform == PlatformID.Win32NT)
                     {
-                        var windowsDefaultDirectory = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData), @"Datadog .NET Tracer", "logs");
+                        // On Nano Server, this returns "", so we fallback to reading from the env var set in the base image instead
+                        // - https://github.com/dotnet/runtime/issues/22690
+                        // - https://github.com/dotnet/runtime/issues/21430
+                        // - https://github.com/dotnet/runtime/pull/109673
+                        // If _that_ fails, we just hard code it to "C:\ProgramData", which is what the native components do anyway
+                        var programData = Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData);
+                        if (string.IsNullOrEmpty(programData))
+                        {
+                            programData = Environment.GetEnvironmentVariable("ProgramData");
+                            if (string.IsNullOrEmpty(programData))
+                            {
+                                programData = @"C:\ProgramData";
```

Comment:
is it ok to assume the C drive? I gues for 99% of cases but 🤷🏻

Reply:
> is it ok to assume the C drive? I gues for 99% of cases but 🤷🏻

Yeah, I had the same thought, but it's what we do on the _native_ side, so the answer is _probably_ yes? We could try getting the `SystemDrive` env var, but if `ProgramData` failed, I'm guessing `SystemDrive` would fail too, so we'll probably end up in the same situation either way? Ultimately, if we can't create the folder, we write to a temp path instead so, meh?